app-android/src/main/java/com/ustadmobile/port/android/view/ClazzListFragment.kt

```diff
@@ -140,4 +165,154 @@ class ClazzListFragment(): UstadListViewFragment<Clazz, ClazzWithListDisplayDeta
     }
 
 
+}
+
+@Composable
+private fun ClazzListScreen(
+    uiState: ClazzListUiState = ClazzListUiState(),
+    onClickClazz: (Clazz) -> Unit = {},
+    onClickSort: () -> Unit = {},
+    onClickFilterChip: (MessageIdOption2) -> Unit = {},
+) {
+
+    LazyVerticalGrid(
+        modifier = Modifier.fillMaxSize(),
+
+        // 600 width of the smallest iPad,
+        // subtracted 20 = horizontal padding & space between cards,
+        // half of 580 is 290
+        columns = GridCells.Adaptive(290.dp)
```

Comment:
This is the right way to do the layout.

Material design uses an 8.px grid, everything should normally be in multiples of 8 dp. 

We want 16dp between the card and edge of screen, and 16 dp between each card. Therefor each card should have 8dp margin itself, and left/right of the grid should have 8.dp margin. 

Card width = (600-16dp horizontal padding) = 584. Divide by 2: card width = 292dp.